pkg/upgrade/control_plane.go

```diff
@@ -673,18 +638,175 @@ func (u *ControlPlaneUpgrader) shouldSkipMachine(log logr.Logger, machine *clust
 	return false
 }
 
+// isHealthy verifies if a machine is healthy.
+// Conditions for health:
+// - the machine should have a ProviderID
+// - the machine should have a node associated with the ProviderID
+// - the node should be Ready
+// - the node should have the LabelNodeRoleMaster
+func (u *ControlPlaneUpgrader) isHealthy(machine *clusterv1.Machine) error {
+	ctx := context.TODO()
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
```

Comment:
Yeah, I wasn't sure what time out to put here since these polling "can" effectively add 15 mins (by default) to the overall upgrade. I wasn't sure of adding other timeout configurations but I agree, I was a bit uncomfortable hardcoding timeouts. I'll switch to `u.machineTimeout` for now.